### PR TITLE
[LTC] Use functools.lru_cache instead

### PR DESCRIPTION
**Pull request by alanwaketan**
Use functools.lru_cache instead of functools.cache given the TorchBench CI only supports Python 3.8.

### CODE REVIEW

File: lazy_tensor_core/lazy_bench.py
Comment (wconstab) [inline lazy_tensor_core/lazy_bench.py:59]: now I'm wondering why I used cache in the first place. Do you know what the difference is, and did you confirm the csv file still has the correct data before/after the change?

Comment (alanwaketan) [reply lazy_tensor_core/lazy_bench.py:59]: No, I didn't... I was trying to move fast and rely on the CI.

Comment (wconstab) [reply lazy_tensor_core/lazy_bench.py:59]: ok. well if it works then I am ok with the change. ill stamp to unblock but i'm assuming you'll make sure it's working before land.

**Review by wconstab — APPROVED** [review]

File: lazy_tensor_core/lazy_bench.py
Comment (alanwaketan) [reply lazy_tensor_core/lazy_bench.py:59]: Yea, it should work as suggested here: https://docs.python.org/3/library/functools.html#functools.cache. But I will verify that before landing.

**Comment by alanwaketan** [comment]
@wconstab We have the results of the script here: https://github.com/pytorch/pytorch/actions/runs/1580797565. That PR #69650 cherry-picked this one. Is it safe to merge now?

**Comment by wconstab** [comment]
> @wconstab We have the results of the script here: https://github.com/pytorch/pytorch/actions/runs/1580797565. That PR #69650 cherry-picked this one. Is it safe to merge now?

You mean the results in that link includes .lru_cache?  I think the important part is to verify that the output .csv file (e.g. lazy_overheads..) contains many rows, for all the different models that ran.

**Comment by alanwaketan** [comment]
> > @wconstab We have the results of the script here: https://github.com/pytorch/pytorch/actions/runs/1580797565. That PR #69650 cherry-picked this one. Is it safe to merge now?
> 
> You mean the results in that link includes .lru_cache? I think the important part is to verify that the output .csv file (e.g. lazy_overheads..) contains many rows, for all the different models that ran.

I guess so?
Screen Shot 2021-12-15 at 1 51 37 PM

**Pull request merged**
